File: business-easy-core/src/main/java/com/businesseasy/core/entities/TxEntity.java
package com.businesseasy.core.entities;

import com.businesseasy.core.common.enums.TxType;
import lombok.*;

import javax.persistence.*;
import java.math.BigDecimal;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "tx")
public class TxEntity extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "amount")
    BigDecimal amount;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "from_account_id")
    AccountEntity fromAccount;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "to_account_id")
    AccountEntity toAccount;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "people_id")
    PeopleEntity peopleEntity;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "document_id")
    DocumentEntity documentEntity;

    @Column(name = "ref")
    String ref;

    @Column(name = "type")
    @Enumerated(EnumType.STRING)
    TxType type;

    @Column(name = "meta")
    String meta;
}

File: easiness-ui/app/server/entity/tx.entity.ts
import * as Big from "big.js";
import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from "typeorm";
import { BigColumn } from "../utils/decorators";
import { Account } from "./account.entity";
import { Base } from "./base.entity";
import { Document } from "./document.entity";
import { People } from "./people.entity";


export enum TxType {
    INCOME = "INCOME",
    EXPENSE = "EXPENSE",
    BANK_TRANSFER = "BANK_TRANSFER"
}

@Entity({ name: 'tx' })
export class Tx extends Base {
    @PrimaryGeneratedColumn()
    id: number

    @BigColumn()
    amount: Big

    @ManyToOne(() => Account)
    fromAccount: Account
    @Column({ nullable: true })
    fromAccountId!: number

    @ManyToOne(() => Account)
    toAccount: Account
    @Column({ nullable: true })
    toAccountId!: number

    @ManyToOne(() => Document)
    document: Document
    @Column({ nullable: true })
    documentId!: number

    @ManyToOne(() => People)
    people: People
    @Column({ nullable: true })
    peopleId: number

    @Column({ nullable: true })
    ref!: string

    @Column()
    type: TxType

    @Column({ nullable: true })
    meta!: string

    @Column({ nullable: true })
    description!: string

}